miasm2/arch/mep/sem.py

```diff
@@ -236,23 +247,18 @@ def add3(ir, instr, reg_dst, reg_src, reg_or_imm):
 
 @sbuild.parse
 def add(arg1, arg2):
-    """ADD - add a register and an immediate"""
+    """ADD - Add a register and an immediate."""
 
     # Rn <- Rn + SignExt(imm6)
     arg1 = arg1 + arg2.signExtend(32)
 
 
 @sbuild.parse
 def advck3(r0, rn, rm):
-    """ADVCK3 - check addition overflow"""
+    """ADVCK3 - Check addition overflow."""
 
     # if(Overflow(Rn+Rm)) R0<-1 else R0<-0 (Signed)
-
-    # Get the register storing the highest value
-    max_rn_rm = rn.zeroExtend(64) if ">"(rn, rm) else rm.zeroExtend(64)
-
-    # Check for an overflow
-    r0 = i32(1) if "<"(rn.zeroExtend(64) + rm.zeroExtend(64), max_rn_rm) else i32(0)
+    r0 = i32(1) if compute_u_inf(i64(0xFFFFFFFF), rn.zeroExtend(64) + rm.zeroExtend(64)) else i32(0)
```

Comment:
It seems that you're testing `0xFFFFFFFF < rn.zeroExtend + rm.zeroExtend`

You may want to do the opposite?

Reply:
My bad, please ignore this remark